emu_mps/mps_backend.py

```diff
@@ -54,8 +60,26 @@ def run(self) -> Results:
 
         impl = create_impl(self._sequence, self._config)
         impl.init()  # This is separate from the constructor for testing purposes.
+        results = self._run(impl)
+        if not self._config.optimise_interaction_matrix:
+            return results
+
+
+        inv_perm = opmat.invert_permutation(impl.opt_perm)
```

Comment:
I would put lines 68 -> 80 into an auxiliary function

Reply:
what if the user didn't ask for storing bitstrings in results?

Reply:
what about
- computing `self.inv_perm` and storing it as class member
- doing this inverse permutation of bitstrings on-the-fly in `fill_results`?

Reply:
shouldn't you ALSO permute the state and hamiltonian that are generically passed to callbacks?
bitstrings are not even the only order-dependent default observable so far: you also have qubit occupation!